File: languages/lua/web3_client.lua
local http = require("socket.http")
local ltn12 = require("ltn12")
local json = require("json")

local RPC_URL = "https://mainnet.infura.io/v3/YOUR_KEY"

local function rpcCall(method, params)
    local request = {
        jsonrpc = "2.0",
        id = 1,
        method = method,
        params = params or {}
    }

    local requestBody = json.encode(request)
    local responseBody = {}

    local result, statusCode = http.request{
        url = RPC_URL,
        method = "POST",
        headers = {
            ["Content-Type"] = "application/json",
            ["Content-Length"] = #requestBody
        },
        source = ltn12.source.string(requestBody),
        sink = ltn12.sink.table(responseBody)
    }

    if result then
        local response = json.decode(table.concat(responseBody))
        return response.result
    end
    return nil
end

local function hexToNumber(hex)
    return tonumber(hex, 16)
end

local function getBlockNumber()
    local result = rpcCall("eth_blockNumber")
    return hexToNumber(result)
end

local function getBalance(address)
    local result = rpcCall("eth_getBalance", {address, "latest"})
    local wei = hexToNumber(result)
    return wei / 1e18
end

local function getGasPrice()
    local result = rpcCall("eth_gasPrice")
    local wei = hexToNumber(result)
    return wei / 1e9
end

-- Main
print("Current block: " .. getBlockNumber())
print("Vitalik balance: " .. getBalance("0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045") .. " ETH")
print("Gas price: " .. getGasPrice() .. " Gwei")

File: languages/perl/web3_client.pl
#!/usr/bin/env perl
use strict;
use warnings;
use LWP::UserAgent;
use JSON;
use Math::BigInt;

my $RPC_URL = 'https://mainnet.infura.io/v3/YOUR_KEY';

sub rpc_call {
    my ($method, $params) = @_;

    my $ua = LWP::UserAgent->new;
    my $json = JSON->new;

    my $request = {
        jsonrpc => '2.0',
        id => 1,
        method => $method,
        params => $params || []
    };

    my $response = $ua->post(
        $RPC_URL,
        Content_Type => 'application/json',
        Content => $json->encode($request)
    );

    if ($response->is_success) {
        my $data = $json->decode($response->content);
        return $data->{result};
    }
    return undef;
}

sub hex_to_number {
    my ($hex) = @_;
    $hex =~ s/^0x//;
    return Math::BigInt->from_hex($hex)->numify();
}

sub get_block_number {
    my $result = rpc_call('eth_blockNumber');
    return hex_to_number($result);
}

sub get_balance {
    my ($address) = @_;
    my $result = rpc_call('eth_getBalance', [$address, 'latest']);
    my $wei = hex_to_number($result);
    return $wei / 1e18;
}

sub get_gas_price {
    my $result = rpc_call('eth_gasPrice');
    my $wei = hex_to_number($result);
    return $wei / 1e9;
}

# Main
my $block = get_block_number();
print "Current block: $block\n";

my $balance = get_balance('0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045');
printf "Vitalik balance: %.4f ETH\n", $balance;

my $gas = get_gas_price();
printf "Gas price: %.2f Gwei\n", $gas;
